src/mastic.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{
+    codec::{CodecError, Decode, Encode, ParameterizedDecode},
+    field::FieldElement,
+    flp::{
+        szk::{Szk, SzkProofShare},
+        Type,
+    },
+    vdaf::{
+        poplar1::Poplar1AggregationParam,
+        xof::{Seed, Xof},
+        Aggregatable, Client, Vdaf, VdafError,
+    },
+    vidpf::{
+        Vidpf, VidpfInput, VidpfKey, VidpfPublicShare, VidpfServerId, VidpfValue, VidpfWeight,
+    },
+};
+use std::{fmt::Debug, io::Cursor, marker::PhantomData};
+/// The main struct implementing the Mastic VDAF.
+/// Composed of a shared zero knowledge proof system and a verifiable incremental
+/// distributed point function.
+#[derive(Clone, Debug)]
+pub struct Mastic<T, P, const SEED_SIZE: usize>
+where
+    T: Type,
+    P: Xof<SEED_SIZE>,
+{
+    algorithm_id: u32,
+    szk: Szk<T, P, SEED_SIZE>,
+    pub(crate) vpf: Vidpf<VidpfWeight<T::Field>, 16>,
```

Comment:
I think it's best to avoid introducing a new acronym.
```suggestion
    pub(crate) vidpf: Vidpf<VidpfWeight<T::Field>, 16>,
```